File: 프로그래머스/3/43162. 네트워크/네트워크.js
function dfs(computers, visited, node) {
    visited[node] = true;
    for (let i = 0; i < computers[node].length; i++) {
        if (computers[node][i] && !visited[i]) {
            dfs(computers, visited, i)
        }
    }
}

function solution(n, computers) {
    let answer = 0;
    const visited = Array.from({length: n}, () => false)
    
    for (let i = 0; i < n; i++) {
        if (!visited[i]) {
            dfs(computers, visited, i)
            answer += 1;
        }
    }
    
    return answer;
}

File: 프로그래머스/3/43162. 네트워크/네트워크.py
def dfs(computers, visited, idx):
    visited[idx] = True
    
    for i in range(len(computers[idx])):
        if i == idx:
            continue
        
        if not visited[i] and computers[idx][i] == 1:
            dfs(computers, visited, i)
    

def solution(n, computers):
    answer = 0
    visited = [False] * n
    
    for i in range(n):
        if not visited[i]:
            dfs(computers, visited, i)
            answer += 1
    
    return answer